portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -71,18 +83,29 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       if(isStringValid(commenterValue)){
         commenter = new Text(commenterValue);
       } else {
-        commenter = new Text("Anonymous");
+        commenter = new Text(email);
       }
       
+      
+
       long timestamp = System.currentTimeMillis();
+
+      String id = userService.getCurrentUser().getUserId();
       
       Entity commentEntity = new Entity("Comment");
       commentEntity.setProperty("comment", comment);
       commentEntity.setProperty("commenter", commenter);
+      commentEntity.setProperty("email", email);
       commentEntity.setProperty("timestamp", timestamp);
 
+      Entity userEntity = new Entity("UserInfo", id);
+      userEntity.setProperty("id", id);
```

Comment:
Is it okay that you'll be storing multiple entities for each ID? If I post 5 messages, won't my UserInfo be recorded 5 different times?

(It's okay to fix this in a follow-up PR if you end up needing it.)

Reply:
It wouldn't store multiple entities for each id because I'm setting the entity key in the constructor by passing in `id`.  The `put` method in `datastore.put(userEntity)` overwrites an entity if it already exists in the datastore (https://cloud.google.com/datastore/docs/concepts/entities#creating_an_entity). Because I'm manually setting the entity key for a user, multiple posts from the same user will find and overwrite the same user entity.
It was recording multiple entities for each ID before I found out that datastore compares the entity keys to know if an entity already exists and that I could manually set the keys.